gpslogger/src/main/AndroidManifest.xml

```diff
@@ -80,7 +80,12 @@
                 <data android:scheme="com.mendhak.gpslogger"
                     android:path="/oauth2googledrive"/>
             </intent-filter>
-
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW"/>
+                <category android:name="android.intent.category.DEFAULT"/>
+                <category android:name="android.intent.category.BROWSABLE"/>
+                <data android:host="oauth2openstreetmap" android:scheme="gpslogger" />
```

Comment:
Try to change this to follow the recommended scheme, `com.mendhak.gpslogger://oauth2openstreetmap` in the OSM settings page.